packages/usa-modal/src/index.js

```diff
@@ -278,51 +279,50 @@ const cleanUpModal = (baseComponent) => {
   modalWrapper.parentElement.removeChild(modalWrapper);
 };
 
-modal = {
-  init(root) {
-    selectOrMatches(MODAL, root).forEach((modalWindow) => {
-      const modalId = modalWindow.id;
-      setUpModal(modalWindow);
-
-      // this will query all openers and closers including the overlay
-      document
-        .querySelectorAll(`[aria-controls="${modalId}"]`)
-        .forEach((item) => {
-          // Turn anchor links into buttons because of
-          // VoiceOver on Safari
-          if (item.nodeName === "A") {
-            item.setAttribute("role", "button");
-            item.addEventListener("click", (e) => e.preventDefault());
-          }
-
-          // Can uncomment when aria-haspopup="dialog" is supported
-          // https://a11ysupport.io/tech/aria/aria-haspopup_attribute
-          // Most screen readers support aria-haspopup, but might announce
-          // as opening a menu if "dialog" is not supported.
-          // item.setAttribute("aria-haspopup", "dialog");
-
-          item.addEventListener("click", toggleModal);
-        });
-    });
-  },
-  teardown(root) {
-    selectOrMatches(MODAL, root).forEach((modalWindow) => {
-      cleanUpModal(modalWindow);
-      const modalId = modalWindow.id;
-
-      document
-        .querySelectorAll(`[aria-controls="${modalId}"]`)
-        .forEach((item) => item.removeEventListener("click", toggleModal));
-    });
-  },
-  focusTrap: null,
-  toggleModal,
-  on(root) {
-    this.init(root);
-  },
-  off(root) {
-    this.teardown(root);
-  },
-};
+modal = behavior(
+  {},
+  {
+    init(root) {
+      selectOrMatches(MODAL, root).forEach((modalWindow) => {
+        const modalId = modalWindow.id;
+        if (modalId) {
+          setUpModal(modalWindow);
+        }
+
+        // this will query all openers and closers including the overlay
+        document
+          .querySelectorAll(`[aria-controls="${modalId}"]`)
+          .forEach((item) => {
+            // Turn anchor links into buttons because of
+            // VoiceOver on Safari
+            if (item.nodeName === "A") {
+              item.setAttribute("role", "button");
+              item.addEventListener("click", (e) => e.preventDefault());
```

Comment:
Do we need this event listener for "click" if we're calling it in line 309?

Reply:
It looks like the anchor tags get rebuilt with the modal ID as it's `href`. Removing this line will cause the page to reload with the modalId as a hash in the URL.

I think I'd like to move the comment for added clarity. Something like:

```js
          // Query all openers and closers including the overlay
          selectOrMatches(`[aria-controls="${modalId}"]`, root).forEach((modalTrigger) => {
            if (modalTrigger.nodeName === "A") {
              // Turn anchor links into buttons for screen readers
              modalTrigger.setAttribute("role", "button");

              // Prevent modal triggers from acting like links
              modalTrigger.addEventListener("click", (e) => e.preventDefault());
            }
```

Reply:
It looks like the anchor tags get rebuilt with the modal ID as it's `href`. Removing this line will cause the page to reload with the modalId as a hash in the URL.

I think I'd like to move the comment for added clarity. Something like:

```js
          // Query all openers and closers including the overlay
          selectOrMatches(`[aria-controls="${modalId}"]`, root).forEach((modalTrigger) => {
            if (modalTrigger.nodeName === "A") {
              // Turn anchor links into buttons for screen readers
              modalTrigger.setAttribute("role", "button");

              // Prevent modal triggers from acting like links
              modalTrigger.addEventListener("click", (e) => e.preventDefault());
            }
```

Reply:
Completed in [24a90b7](https://github.com/uswds/uswds/pull/5208/commits/24a90b76553b0c06ff32fa843bac2f67b28833a1)